h2/src/main/org/h2/table/MetaTable.java

```diff
@@ -1688,7 +1688,7 @@ private boolean hideTable(Table table, Session session) {
                         // CONSTRAINT_NAME
                         identifier(constraint.getName()),
                         // CONSTRAINT_TYPE
-                        constraintType.toString(),
+                        constraintType.getSqlName(),
```

Comment:
With this change you get another regression. You should use something like
```Java
constraintType == Constraint.Type.PRIMARY_KEY ? constraintType.getSqlName() : constraintType.name()
```
because `getSqlName()` returns standard SQL name. Standard name is used, for example, in standard meta table `INFORMATION_SCHEMA.CONSTRAINTS.TABLE_CONSTRAINTS`. But H2's own `INFORMATION_SCHEMA.CONSTRAINTS.CONSTRAINTS` use another naming convention. The difference is with `FOREIGN KEY` constraints. `getSqlName()` returns `FOREIGN KEY`, but `REFERENTIAL` was returned here.

Reply:
I see, thanks for the explanation.